### PR TITLE
HttpContext: do not lowercase type when resolving dynamic converter

### PR DESCRIPTION
**Pull request by jorrit**
Fixes strongloop/loopback#906

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: test/http-context.test.js [hunk 1]
```diff
@@ -74,6 +75,24 @@ describe('HttpContext', function() {
...
+      });
+
+      it('should coerce dynamic type with string prop into object', givenMethodExpectArg({
+        type: 'CustomType',
+        input: JSON.stringify({ stringProp: 'string' }),
+        expectedValue: { stringProp: 'string' }
+      }));
+
+      it('should coerce dynamic type with int prop into object', givenMethodExpectArg({
+        type: 'CustomType',
```
Comment (ritch) [inline test/http-context.test.js]: Add a test that breaks before this patch... eg `type: 'customType'`

**Comment by ritch** [comment]
Other than a missing tests that fails before this patch this LGTM

File: test/http-context.test.js (hunk 1 above)
Comment (jorrit) [inline test/http-context.test.js]: I believe this test will break before the patch. There is no functional difference between `CustomType` and `customType` in this regard, both will fail. Only `customtype` will not fail at the moment.

**Comment by jorrit** [comment]
Can you explain what is wrong with the tests that I added? They will fail when the code change is not there.

**Pull request merged**